pkg/config/plugins/resources/postgres/config.go

```diff
@@ -22,6 +21,51 @@ type PostgresStoreConfig struct {
 	DbName string `yaml:"dbName" envconfig:"kuma_store_postgres_db_name"`
 	// Connection Timeout to the DB in seconds
 	ConnectionTimeout int `yaml:"connectionTimeout" envconfig:"kuma_store_postgres_connection_timeout"`
+	// TLS settings
+	TLS TLSPostgresStoreConfig `yaml:"tls"`
+}
+
+// Modes available here https://godoc.org/github.com/lib/pq
+type TLSMode string
+
+const (
+	Disable TLSMode = "disable"
+	// Always TLS (skip verification)
+	VerifyNone TLSMode = "verifyNone"
+	// Always TLS (verify that the certificate presented by the server was signed by a trusted CA)
+	VerifyCa TLSMode = "verifyCa"
+	// Always TLS (verify that the certification presented by the server was signed by a trusted CA and the server host name matches the one in the certificate)
+	VerifyFull TLSMode = "verifyFull"
+)
+
+type TLSPostgresStoreConfig struct {
+	// Mode of TLS connection. Available values (disable, verifyNone, verifyCa, verifyFull)
+	Mode TLSMode `yaml:"mode" envconfig:"kuma_store_postgres_tls_mode"`
+	// Path to TLS Certificate of the client. Used in require, verifyCa and verifyFull modes
+	CertPath string `yaml:"certPath" envconfig:"kuma_store_postgres_tls_cert_path"`
+	// Path to TLS Key of the client. Used in verifyNone, verifyCa and verifyFull modes
+	KeyPath string `yaml:"keyPath" envconfig:"kuma_store_postgres_tls_key_path"`
+	// Path to the root certificate. Used in verifyCa and verifyFull modes.
+	CAPath string `yaml:"caPath" envconfig:"kuma_store_postgres_tls_ca_path"`
+}
+
+func (s TLSPostgresStoreConfig) Sanitize() {
+}
+
+func (s TLSPostgresStoreConfig) Validate() error {
+	switch s.Mode {
+	case VerifyFull:
+		fallthrough
+	case VerifyCa:
+		if s.CAPath == "" {
+			return errors.New("CAPath cannot be empty")
+		}
+	case VerifyNone:
+	case Disable:
+	default:
+		return errors.Errorf("invalid mode: %s", s.Mode)
+	}
+	return nil
```

Comment:
Let's verify that if `Cert` is non-empty then so must be the `Key`. And vice versa